### PR TITLE
Optical flow: compute velocity from corrected flow data for logging

### PR DESCRIPTION
**Pull request by bresch**
This is really useful to align the flow with the IMU data and verify that the compensation is properly done

update to https://github.com/PX4/ecl/pull/920

includes:
- logging of compensated optical flow data converted to velocity measurement in body and local frames
- logging of un/compensated optical flow data for rate motion and synchronized gyro data
- adjust scaling value of pmw3901 data (was off by 30%)
- adjust delay parameter for pmw3901 for perfect synchronization with gyro data

Bench tested with pmw3901 optical flow sensor and Lanbao PSK-CM8JL65 range finder:
*Note that I recently renamed the fields in `optical_flow_vel` so the signals in the screenshot below don't have the correct names*
flow_scaled_to_gyro

### CODE REVIEW

File: src/drivers/optical_flow/pmw3901/PMW3901.cpp [hunk 1]
```diff
@@ -324,8 +324,8 @@ PMW3901::RunImpl()
 		return;
 	}
 
-	delta_x = (float)_flow_sum_x / 500.0f;		// proportional factor + convert from pixels to radians
-	delta_y = (float)_flow_sum_y / 500.0f;		// proportional factor + convert from pixels to radians
+	delta_x = (float)_flow_sum_x / 385.0f;		// proportional factor + convert from pixels to radians
```
Comment (dagar) [inline src/drivers/optical_flow/pmw3901/PMW3901.cpp:327]: Likely applies to the paw3902 as well.

File: msg/CMakeLists.txt [hunk 2]
```diff
@@ -163,6 +163,7 @@ set(msg_files
 	wheel_encoders.msg
 	wind_estimate.msg
 	yaw_estimator_status.msg
+	optical_flow_vel.msg
```
Comment (dagar) [inline msg/CMakeLists.txt:166]: Maybe prefix this message with `estimator_` so we know where it's coming from?

File: src/modules/ekf2/EKF2.cpp [hunk 3]
```diff
@@ -1333,6 +1336,21 @@ void EKF2::publish_wind_estimate(const hrt_abstime &timestamp)
 	}
 }
 
+void EKF2::publish_optical_flow_vel(const hrt_abstime &timestamp)
+{
+	optical_flow_vel_s flow_vel{};
+	flow_vel.timestamp_sample = timestamp;
+
+	_ekf.getFlowVelBody().copyTo(flow_vel.vel_body);
```
Comment (dagar) [inline src/modules/ekf2/EKF2.cpp:1344]: Could this be similar to `vehicle_visual_odometry_aligned` where it only publishes in the same cycle that new data was received, but after the EKF update?

File: src/modules/logger/logged_topics.cpp [hunk 4]
```diff
@@ -109,6 +109,7 @@ void LoggedTopics::add_default_topics()
 	add_topic("vehicle_status_flags");
 	add_topic("vtol_vehicle_status", 200);
 	add_topic("yaw_estimator_status", 200);
+	add_topic("optical_flow_vel", 200);
```
Comment (dagar) [inline src/modules/logger/logged_topics.cpp:112]: Generally I try to keep anything like this sorted alphabetically. Lists of topics, headers included, etc.

File: src/drivers/optical_flow/pmw3901/PMW3901.cpp (hunk 1 above)
Comment (bresch) [reply src/drivers/optical_flow/pmw3901/PMW3901.cpp:327]: @dagar I thought the same but don't have one to verify that; Are you sure that the scaling is the same?

File: src/modules/ekf2/EKF2.cpp (hunk 3 above)
Comment (bresch) [reply src/modules/ekf2/EKF2.cpp:1348]: Not sure if it's worth doing this because the optical flow update rate is probably faster than the EKF update rate

Comment (dagar) [reply src/modules/ekf2/EKF2.cpp:1348]: It's more of a log visibility (and minor optimization) thing, only having `estimator_optical_flow_vel` when you actually had optical_flow and were using it.

Getting the optical_flow data rate within the EKF update rate is one of the other reasons I was going to pipe it through the sensors module first.

**Comment by bresch** [comment]
@dagar I addressed your comments, any other remarks?

**Comment by dagar** [comment]
> @dagar I addressed your comments, any other remarks?

Thanks, good to go from my perspective.

**Review by dagar — DISMISSED** [review]

**Comment by bresch** [comment]
unrelated SITL failure

**Comment by bresch** [comment]
@dagar I just squashed two commits together to make the history cleaner, could you approve again please?

**Review by dagar — DISMISSED** [review]

File: src/modules/logger/logged_topics.cpp [hunk 5]
```diff
@@ -60,6 +60,7 @@ void LoggedTopics::add_default_topics()
 	add_topic("estimator_innovation_test_ratios", 200);
 	add_topic("estimator_innovation_variances", 200);
 	add_topic("estimator_innovations", 200);
+	add_topic("estimator_optical_flow_vel");
```
Comment (dagar) [inline src/modules/logger/logged_topics.cpp:63]:
```suggestion
	add_topic("estimator_optical_flow_vel", 200);
```

Limit default logging rate.

**Pull request merged**